src/main/java/org/launchcode/sprinklespre/models/Ingredient.java

```diff
@@ -0,0 +1,68 @@
+package org.launchcode.sprinklespre.models;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.ManyToMany;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+@Entity
+public class Ingredient extends AbstractEntity{
+
+    @ManyToMany
+    private List<Module> modules = new ArrayList<>();
+
+    // Amount of the ingredient that is needed for the module(recipe)
+    private Double amountNeeded;
+
+    private String unit;
+
+    // What type of ingredient is it (vegetable, fruit, protein, etc). Meant to make searching easier.
+    private Set<String> tags;
+
```

Comment:
I wonder if we could do this at the module level as a "main" ingredient? So a module (recipe) get's tagged from a Set as having a main ingredient that is grain, vegetable, protein, fruit, other? I'm definitely second guessing myself here as the data gets more complicated.

Reply:
I suppose we could, but what if a recipe has several "main" ingredients, such as a pasta dish with vegetables and protein in it?

Reply:
This feels like a very librarian conversation haha. I think 'main' is subjective, and we'd leave it up to the person creating the module to pick. In your example, I'd pick 'grain' for main ingredient, but someone else could pick something else from the list. If we tagged each ingredient, is the idea that those tags would then be added _at the module or course level_ for searching/filtering? That actually makes sense, but in the absence of a classifying algorithm (this might exist somewhere already but my palms are getting sweaty just thinking about it) it seems taxing for the user to go through and tag each ingredient. But I do think tags make sense at the module or course level.

Reply:
Upon further thought, yeah, in this case, it makes more sense at the module level.